packages/offix-datastore/src/storage/Storage.ts

```diff
@@ -1,57 +1,76 @@
 import { PushStream, ObservablePushStream } from "../utils/PushStream";
-import { Model, PersistedModel } from "../models";
 import { PredicateFunction } from "../predicates";
 import { createDefaultStorage } from "./adapters/defaultStorage";
 import { generateId } from "./core";
 
+/**
+ * This interface defines the API that is required
+ * from any Device specific Storage implementation
+ * It defines a CRUD interface that a
+ * Device Specific Implementation must support
+*/
 export interface IStorageAdapter {
-    save(model: PersistedModel): Promise<PersistedModel>;
-    query(modelName: string, predicate?: PredicateFunction): Promise<PersistedModel | PersistedModel[]>;
-    update(model: PersistedModel): Promise<PersistedModel>;
-    remove(model: PersistedModel, predicate?: PredicateFunction): Promise<PersistedModel | PersistedModel[]>;
+    save(storeName: string, input: any): Promise<any>;
+    query(storeName: string, predicate?: PredicateFunction): Promise<any | any[]>;
+    update(storeName: string, input: any, predicate?: PredicateFunction): Promise<any>;
+    remove(storeName: string, predicate?: PredicateFunction): Promise<any | any[]>;
```

Comment:
Do you mind adding JSDoc comments for each of the interface fields please. Just helps document the code better.

Here's an exmaple:
https://github.com/aerogear/offix/blob/bcb2b689817536876c9166d6ef5be5a3e3ec6d3b/packages/offix-cache/src/createMutationOptions.ts#L61